### PR TITLE
extensions e2e test breakage on Windows: fix running the dummy.cmd on windows by using / instead of \.

### LINKED ISSUE #4246: use of the extension protocol handler is failing on windows
### Actual Behavior

extensions-protocol.e2e.spec.ts:104:3 › Extensions protocol handler › build and install testing extension is failing on windows 11 by running the e2e tests via CI.

### Steps to Reproduce

Triggering e2e tests to run via CI on windows 11.

### Result

Test is showing the following error:
```
2) extensions-protocol.e2e.spec.ts:104:3 › Extensions protocol handler › build and install testing extension 

    Error: C:\actions-runner\_work\rancher-desktop\rancher-desktop\resources\win32\bin\nerdctl.exe exited with code 1

       at ..\pkg\rancher-desktop\utils\childProcess.ts:245

      243 |         message = `${ command } exited with signal ${ signal }`;
      244 |       }
    > 245 |       const error: SpawnError = new Error(message);
          |                                 ^
      246 |
      247 |       Object.defineProperties(error, {
      248 |         [ErrorCommand]: {

        at ChildProcess.<anonymous> (C:\actions-runner\_work\rancher-desktop\rancher-desktop\pkg\rancher-desktop\utils\childProcess.ts:245:33)
```

### Expected Behavior

The test should pass in all platforms

### Additional Information

_No response_

### Rancher Desktop Version

N/A

### Rancher Desktop K8s Version

N/A

### Which container engine are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

windows 11

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

None

### Windows User Only

_No response_

### PR DESCRIPTION
**Pull request by ericpromislow**
Fixes #4246

If we don't convert the backslashes in `C:Program Files\node.exe`  we get error messages like

```
C:Program Files' is not recognized as an internal or external command,·
    operable program or batch file.·
```

The simplest way is to convert the backslashes to forward-slashes. Modern Windows CLIs now
process / as directory separators.

### CODE REVIEW

File: e2e/extensions.e2e.spec.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ import type { BrowserView, BrowserWindow } from 'electron';
 /** The top level source directory, assuming we're always running from the tree */
 const srcDir = path.dirname(path.dirname(__filename));
 const rdctl = getFullPathForTool('rdctl');
+const execPath = process.execPath.replace(/\\/g, '/');
```
Comment (mook-as) [inline e2e/extensions.e2e.spec.ts:28]: Since this is so tricky, maybe add a comment about why we're doing this?

File: e2e/extensions.e2e.spec.ts [hunk 2]
```diff
@@ -202,7 +203,7 @@ test.describe.serial('Extensions', () => {
       test('capturing output', async() => {
         const script = `
           ddClient.extension.host.cli.exec("${ wrapperName }", [
-            "${ process.execPath }", "-e", "console.log(1 + 1)"
+            "${ execPath }", "-e", "console.log(1 + 1)"
           ]).then(({cmd, killed, signal, code, stdout, stderr}) => ({
```
Comment (mook-as) [inline e2e/extensions.e2e.spec.ts:207]: Alternatively:
```suggestion
          ddClient.extension.host.cli.exec("${ wrapperName }", ${
            JSON.stringify([process.execPath, '-e', 'console.log(1 + 1)'])
          }).then(({cmd, killed, signal, code, stdout, stderr}) => ({
```

What you have is fine, and this was more of an intellectual exercise.

**Review by mook-as — CHANGES_REQUESTED** [review]
Please add a comment.

File: e2e/extensions.e2e.spec.ts (hunk 2 above)
Comment (ericpromislow) [reply e2e/extensions.e2e.spec.ts:207]: That looks more complex

**Pull request merged**